src/suricata-plugin.h

```diff
@@ -41,16 +41,30 @@ typedef struct SCPlugin_ {
     void (*Init)(void);
 } SCPlugin;
 
+typedef struct SCPluginFileOptions_ {
+    bool threaded;
+    const char *filename; /* Configured file name */
+    const char *log_dir;
+    const char *append;
```

Comment:
I'm not to keen on seeing file specific things in here.  For the log dir `ConfGet("default-log-dir", ...)` can be used. This is easier now with the libsuricata PR in as you can do a `make install-headers` and you have all the headers available.

The `ConfNode` is already passed in which can be used to discover the filename, and append mode, if needed.

Threaded should be considered separately. The `output` plugin *has* to take this value into consideration right?  If the plugin doesn't support threading, but threaded is true things are going to fail as the threading happens at a higher (or lowever - whatever) level.  Making it a parameter makes it more clear it needs to be considered.

Other thought: could the plugin choose to reject the threaded value and still work OK? I'm guessing not, at the very least it would need to know its multi-threaded to do some lock operation.

Reply:
I'll make the `conf` related changes -- good point.

Regarding threaded -- if threading is requested and the plugin can't support it, then the plugin should "fail" (and not work). Making it an explicit param is a good idea -- with failure if there's a mismatch.